confmap/converter/expandconverter/expand.go

```diff
@@ -25,6 +23,8 @@ type converter struct {
 
 // NewFactory returns a factory for a  confmap.Converter,
 // which expands all environment variables for a given confmap.Conf.
+//
+// Deprecated: [v0.107.0] BASH-style env var expansion is deprecated. Use `${FOO}` or `${env:FOO}` instead.
```

Comment:
I think we should also say that `envprovider` should be used instead.

Reply:
Updated

Reply:
Maybe i'm missing something, but i dont see `envprovider` anywhere

Reply:
Neither do I. @TylerHelmuth did you push your changes?

Reply:
It does in fact help when the changes are pushed